### PR TITLE
Adding user doc for Registry

### PR DESCRIPTION
**Pull request by stuclem**
Adding user doc about how to perform operations in the VIC Registry interface. 

- Fixes https://github.com/vmware/vic-product/issues/57
- Fixes https://github.com/vmware/vic-product/issues/58
- Fixes https://github.com/vmware/vic-product/issues/60
- Fixes https://github.com/vmware/vic-product/issues/56

@hainingzhang and @reasonerjt please can you review these doc changes? It might be easier to see the doc as it will appear in its generated final form, rather than in raw Markdown. The generated HTML doc is posted at https://stuclem.github.io/vic-product/assets/files/html/1.1/vic_dev_ops/using_registry.html

**NOTE**: I have not yet updated the topic on replicating images, so please do not review that one yet.

### CODE REVIEW

File: docs/user_doc/vic_dev_ops/configure_registry.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Configure a Registry #
+
+When you first log in to a new vSphere Integrated Containers Registry instance, you can optionally configure the registry to implement authentication of users by using an LDAP or Active Directory service. You can define who can create projects or register as users, implement certificate verification for image replications, set up a mail server, and set the period of validity for login sessions.
+
+**IMPORTANT**: The option to switch from local user management that uses a database to LDAP authentication is only available while the local database is empty. If you start to populate the database with users and projects, the option to switch to LDAP authentication is disabled. If you want to implement LDAP authentication, you must enable this option when you first log in to a new registry instance, before you create any projects or users. 
+
+With the exception of setting up LDAP authentication after you have already created projects or users, you can change any of the other settings at any time after the initial configuration. This includes changing from LDAP authentication back to local user management that uses a database.
```
Comment (hainingzhang) [inline docs/user_doc/vic_dev_ops/configure_registry.md]: Auth Mode can only be changed when there are no projects or users in registry. So LDAP mode cannot be simply switched back to db mode if there are already uses and projects.

File: docs/user_doc/vic_dev_ops/creating_users_registry.md [hunk 2]
```diff
@@ -1,40 +1,24 @@
-# Creating and Managing Users in vSphere Integrated Containers Registry #
+# Create Users in vSphere Integrated Containers Registry #
 
-##Role Based Access Control
+If you configured vSphere Integrated Containers Registry to use local user management rather than LDAP authentication, you must create user accounts before you can assign users to projects. 
 
-In vSphere Integrated Containers Registry, images are grouped under projects. To access an image, a user should be added as a member into the project of the image. A member can have one of the three roles:  
+If the registry uses LDAP authentication, you do not need to create users manually. Users are automatically added to the registry by the LDAP service.
```
Comment (hainingzhang) [inline docs/user_doc/vic_dev_ops/creating_users_registry.md]: This is not true. Suggest something like: 

If the registry uses LDAP authentication, you cannot create or register new user in the registry. This is due to the fact that users are managed externally by an LDAP server. However, users are required to log in at least once using their LDAP credentials in order to be added to the registry account system. After the user's login, you can assign the user to projects.

File: docs/user_doc/vic_dev_ops/creating_users_registry.md [hunk 3]
```diff
@@ -1,40 +1,24 @@
...
 
-* **SysAdmin**: "SysAdmin" has the most privileges. In addition to the privileges mentioned above, "SysAdmin" can also list all projects, set an ordinary user as administrator and delete users. The public project "library" is also owned by the administrator.  
-* **Anonymous**: When a user is not logged in, the user is considered as an "anonymous" user. An anonymous user has no access to private projects and has read-only access to public projects.  
+   Use the `admin` account, or an account with Administrator privileges. If the vSphere Integrated Containers appliance uses a different port for vSphere Integrated Containers Registry, replace 443 with the appropriate port.
+2. Expand **Administration** on the left, click **Users**, then click the **+ Users** button.
+3. Enter a user name, email address, and the user's full name.
 
-##User account
-vSphere Integrated Containers Registry supports two authentication modes:  
+   The user name and email address must be unique in this registry instance. The email address and the user's full name are for use in password reminder emails.
```
Comment (hainingzhang) [inline docs/user_doc/vic_dev_ops/creating_users_registry.md]: maybe "password resetting email" could be better than "password reminder email"

File: docs/user_doc/vic_dev_ops/creating_projects_registry.md [hunk 4]
```diff
@@ -1,30 +1,27 @@
...
+# Create a Project in vSphere Integrated Containers Registry #
 
-##Managing projects
-A project in vSphere Integrated Containers Registry contains all repositories of an application. No images can be pushed to vSphere Integrated Containers Registry before the project is created. RBAC is applied to a project. There are two types of projects in vSphere Integrated Containers Registry:  
+In vSphere Integrated Containers Registry, you group container image repositories in projects. A project contains all of the repositories that an application requires. You cannot push images to vSphere Integrated Containers Registry until you have created a project. 
 
-* **Public**: All users have the read privilege to a public project, it's convenient for you to share some repositories with others in this way.
-* **Private**: A private project can only be accessed by users with proper privileges.  
+**NOTE**: The current version of vSphere Integrated Containers Engine does not support `docker push`. To push images to vSphere Integrated Containers Registry, use regular Docker. You can then pull the images from the registry to a vSphere Integrated Containers Engine virtual container host (VCH).
```
Comment (hainingzhang) [inline docs/user_doc/vic_dev_ops/creating_projects_registry.md]: use regular Docker.  ->  use regular Docker client.

File: docs/user_doc/vic_dev_ops/manage_project_members.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+**Procedure**
+
+1. Log in to the vSphere Integrated Containers Registry interface at https://<i>vic_appliance_address</i>:443.
+
+   Use the `admin` account, an account with the system-wide Administrator role, or an account that has the Project Admin role for this project. If the vSphere Integrated Containers appliance uses a different port for vSphere Integrated Containers Registry, replace 443 with the appropriate port.
+2. Click **Projects** on the left, click a project name, and click **Members**.
+3. In the list of project members, click the 3 vertical dots next to a user name and select an option.
+
+   - To assign the member to a different role in the project, select **Project Admin**, **Developer**, or **Guest**.
+   - To remove the member from the project, select **Delete**.
```
Comment (hainingzhang) [inline docs/user_doc/vic_dev_ops/manage_project_members.md]: please see if this info is needed:

The deletion of project or images are soft delete, i.e. the physical files of the deleted images are not actually deleted. Those files are recycled and their storage space is freed up during a garbage collection process.

File: docs/user_doc/vic_dev_ops/replicating_images.md [hunk 6]
```diff
@@ -24,15 +24,11 @@ To enable image replication, a policy must first be created. Click "Add New Poli
 
 **Note:** Set **"Verify Remote Cert"** to off according to the [installation guide](installation_guide_ova.md) if the destination uses a self-signed or untrusted certificate.
```
Comment (hainingzhang) [inline docs/user_doc/vic_dev_ops/replicating_images.md]: "Verify Remote Cert " can be set in the admin UI, under Administration->Configuration->replication tab

File: docs/user_doc/vic_dev_ops/assign_admin_role.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
+# Assign the System-Wide Administrator Role to Users # 
+
+You can assign the system-wide administrator role to any user. Users with the administrator role have extra permissions in addition to their project-specific privileges. 
+
+- Browse all projects
```
Comment (hainingzhang) [inline docs/user_doc/vic_dev_ops/assign_admin_role.md]: Browse and manage all projects

File: docs/user_doc/vic_dev_ops/assign_admin_role.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+- Assign the administrator role to other users
+- Delete users
+
```
Comment (hainingzhang) [inline docs/user_doc/vic_dev_ops/assign_admin_role.md]: Other privileges include:
Register a new user
Manage replication rules of a project.
Manage system wide configuration.

File: docs/user_doc/vic_dev_ops/assign_admin_role.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+You have created at least one user.
+
+**Procedure**
+
+1. Log in to the vSphere Integrated Containers Registry interface at https://<i>vic_appliance_address</i>:443.
+
+   Use the `admin` account, or an account with Administrator privileges. If the vSphere Integrated Containers appliance uses a different port for vSphere Integrated Containers Registry, replace 443 with the appropriate port.
+2. Expand **Administration** on the left, and click **Users**.
+7. In the list of users, click the 3 vertical dots next to a user name and select **Enable administrator**.
```
Comment (hainingzhang) [inline docs/user_doc/vic_dev_ops/assign_admin_role.md]: The menu has been changed to 
"Set as Administrator"

and "Revoke Administrator" when the user already has an admin role.

File: docs/user_doc/vic_dev_ops/delete_repository_registry.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
+# Delete a Repository
+
+Deleting a repository involves two steps. First, you delete a repository in vSphere Integrated Containers Registry interface. This is known as soft deletion. You can delete the entire repository or just one tag in the repository. After a soft deletion, the registry no longer manages the repository. However, the repository files remain in the registry storage until you run garbage collection by restarting the registry. 
+
+**IMPORTANT**: The option to activate garbage collection is only available when you deploy the vSphere Integrated Containers appliance. After you have deployed the appliance, you cannot retroactively enable or disable garbage collection.
```
Comment (hainingzhang) [inline docs/user_doc/vic_dev_ops/delete_repository_registry.md]: please check this with Fabio:

After you have deployed the appliance, you cannot retroactively enable or disable garbage collection.

File: docs/user_doc/vic_dev_ops/assign_admin_role.md (hunk 7 above)
Comment (stuclem) [inline docs/user_doc/vic_dev_ops/assign_admin_role.md]: Fixed.

File: docs/user_doc/vic_dev_ops/assign_admin_role.md (hunk 8 above)
Comment (stuclem) [inline docs/user_doc/vic_dev_ops/assign_admin_role.md]: Fixed

File: docs/user_doc/vic_dev_ops/configure_registry.md (hunk 1 above)
Comment (stuclem) [inline docs/user_doc/vic_dev_ops/configure_registry.md]: Fixed

File: docs/user_doc/vic_dev_ops/creating_projects_registry.md (hunk 4 above)
Comment (stuclem) [inline docs/user_doc/vic_dev_ops/creating_projects_registry.md]: Fixed

File: docs/user_doc/vic_dev_ops/creating_users_registry.md (hunk 2 above)
Comment (stuclem) [inline docs/user_doc/vic_dev_ops/creating_users_registry.md]: Fixed

File: docs/user_doc/vic_dev_ops/creating_users_registry.md (hunk 3 above)
Comment (stuclem) [inline docs/user_doc/vic_dev_ops/creating_users_registry.md]: Changed to "The email address and the user's full name are for use in email responses to password reset requests."

File: docs/user_doc/vic_dev_ops/delete_repository_registry.md (hunk 10 above)
Comment (stuclem) [inline docs/user_doc/vic_dev_ops/delete_repository_registry.md]: I just checked with Fabio. All settings can be reconfigured after ther initial deployment, except for the root password. I'll add a topic on reconfiguring the appliance (including activating GC).

File: docs/user_doc/vic_dev_ops/assign_admin_role.md (hunk 9 above)
Comment (stuclem) [inline docs/user_doc/vic_dev_ops/assign_admin_role.md]: Checked in newest build and fixed. I also note that special characters are now allowed in passwords, so updated that too.

File: docs/user_doc/vic_dev_ops/manage_project_members.md (hunk 5 above)
Comment (stuclem) [inline docs/user_doc/vic_dev_ops/manage_project_members.md]: This information is included in the topic on deleting repositories: https://github.com/stuclem/vic-product/blob/registry/docs/user_doc/vic_dev_ops/delete_repository_registry.md

**Comment by stuclem** [comment]
Thanks @reasonerjt, I just added https://github.com/vmware/vic-product/issues/153 to include this. Because Notary support is being added to VIC Engine in 1.2, I thought that even though Notary support is implemented, it was only supported from the next release. Thanks for letting me know!

**Comment by stuclem** [comment]
From @hainingzhang by email:

----

1. In file docs/user_doc/vic_app_dev/configure_docker_client.md

    Restart the Docker daemon.
    `$ sudo systemctl daemon-reload`
    `$ sudo systemctl restart docker`

     [HZ] The above steps may not be required because docker client can find out the newly added `ca.crt` automatically. Please verify.

2. vic_app_dev/build_push_images.md

    Tag the image for uploading to the appropriate project repository in vSphere Integrated Containers Registry.
    `docker tag busybox registry_address/project_name/busybox`

     [HZ] Suggest using tag for busybox, e.g. 

    `docker tag busybox:1.0 registry_address/project_name/busybox:1.0`
    and all subsequent CLI of busybox with tag. 

3. docs/user_doc/vic_dev_ops/replicating_images.md 
    Some terminologies have been changed: 

    - Replication policy -> Replication Rule
    - Destination -> Endpoint

    Please update accordingly.

4. /docs/user_doc/vic_dev_ops/push_pull_images.md
    In SUMMARY.md, broken link to this file, please check if it is still necessary.

**Comment by stuclem** [comment]
I need to merge this branch as it is probably going to cause conflicts with other branches. I will resume the review separately, with Notary.
